devito/ir/clusters/algorithms.py

```diff
@@ -343,24 +343,31 @@ def rule(size, e):
         # Reconstruct the Clusters
         processed = []
         for c in clusters:
+            exprs = c.exprs
```

Comment:
This doesn't look like it gets used before being overwritten in the `for` loop below

Reply:
yes, but I like to stick to the conventional ordering "first we pull/transform/handle the exprs, then the ispace (and its related metadata such as the sub-iterators), then the guards, syncs, properties, etc". Basically the order of the kwargs in the Cluster constructor.